package.json

```diff
@@ -148,18 +149,18 @@
     "protomaps": "1.19.0",
     "raw-loader": "^1.0.0",
     "rc-slider": "^9.7.2",
-    "react": "^16.14.0",
+    "react": "^18.2.0",
     "react-anything-sortable": "^1.5.2",
     "react-color": "^2.19.3",
     "react-datepicker": "0.53.0",
-    "react-dom": "^16.14.0",
+    "react-dom": "^18.2.0",
     "react-ga4": "^2.1.0",
     "react-i18next": "^11.18.0",
     "react-responsive": "^5.0.0",
     "react-select": "^3.1.1",
     "react-swipeable": "^5.1.0",
     "react-transition-group": "^4.3.0",
-    "react-uid": "^2.3.0",
+    "react-uid": "^2.3.3",
     "react-virtual": "~2.3.2",
```

Comment:
Should we update any of these other react-based libraries?

Reply:
IMO there are some dependencies that probably should be upgraded as they list older versions of react as peer dependencies (quick check can be done via `npx check-peer-dependencies --yarn`)
- `@visx/*` should be upgraded to _at least_ v2.10.0, it is the first version that lists react 18 as dependency
- `html-to-react` version used has peer dependency on react 16/17, newer versions are on react 18
- `react-anything-sortable`
- `react-datepicker`
- `react-responsive`
- `react-select`
- `react-swipeable`
- `react-virtual`
- `react-shallow-testutils`

Reply:
It's unlikely that the libraries are incompatible with React 18, and more likely that library authors are using version strings such as
```
"react": "^16.0 || ^17.0"
```

when they should use the standard:
```
"react": "^0.13.0 || ^0.14.0 || >=15"
```

Reply:
A library that definitely needs updating is [mobx-react](https://github.com/mobxjs/mobx/tree/main/packages/mobx-react#readme)

Reply:
@steve9164 there will be issues with libraries listing react in peerDependencies as 
```
"react": "^16.0 || ^17.0"
```
if we are on react 18 after upgrading to the new version of the node which comes with a new npm that automatically installs and checks peerDependencies. You can see the error occurring in https://github.com/TerriaJS/terriajs/pull/6986